### DISCUSSION TITLE
Question: what pattern should be used for DRY values in style keys?

### DISCUSSION THREAD

**Discussion by slightly-askew** [facebook/stylex#611] [Q&A] [answered]
### Describe the feature request

Hey. I've previously assigned media values to a variable (either globally or per-file):

![CleanShot 2024-06-20 at 12 23 30](https://github.com/facebook/stylex/assets/23715191/603e585b-984d-4401-ab05-83ae1af7a17a)

I would happy keeping the magic number value in my .stylex.ts file and importing it, but I can't seem to do that without getting an error either—it needs to be the full variable object. I have responsive tokens in my variables file but keeping arbitrary component styles such as the screenshot example in a external .stylex.ts file feels a little abstract imho.

Is there an idiomatic way to achieve this in Stylex that I have missed?

**Comment by slightly-askew** [comment]
Apologies, missed the discussions tab.

I found the answer from another thread:
![CleanShot 2024-06-21 at 13 57 13](https://github.com/facebook/stylex/assets/23715191/df7136bc-0d22-4328-a3e0-c832440625ba)

It would be amazing to share these queries or magic numbers across files but appreciate it's not possible.

**Comment by Jta26** [comment]
Generally verbosity and repetition are not issues for StyleX because they get de-duplicated anyways, but yeah it would be convenient from a devx perspective to be able to share these viewport values, but a solution to this is to create css variables using `stylex.defineVars()` for things you want to reuse widely.

see: https://stylexjs.com/docs/learn/theming/defining-variables/#using-media-queries

**Comment by nmn** [comment, marked as answer]
This may be relevant: https://github.com/facebook/stylex/issues/238
